README.md

```diff
@@ -1,2 +1,462 @@
+| run\_after           | Command that should run after saving the certificate. See [run hooks](#run-hooks).                | str         | no       | -                       |
+| principal            | Kerberos principal.                                                                               | str         | no       | -                       |
+| provider             | The underlying method used to request and manage the certificate.                                 | str         | no       | *Varies by CA*          |
+
+
+### common_name
+
+If `common_name` is not set the role will attempt to use the first
+value of `dns` or `ip`, respectively, as the default. If `dns` and
+`ip` are also not set, `common_name` will not be included in the certificate.
+
+
+### key_size
+
+Recommended minimal-values for a certificate key size, from different
+organizations, vary across time. In the attempt to provide safe settings,
+the default minimal-value for `key_size` will be increased over time.
+
+If you want your certificates to always keep the same `key_size` when
+renewed, set this variable to the desired value.
+
+
+### key_usage
+
+Valid values for `key_usage` are:
+
+* digitalSignature
+* nonRepudiation
+* keyEncipherment
+* dataEncipherment
+* keyAgreement
+* keyCertSign
+* cRLSign
+* encipherOnly
+* decipherOnly
+
+The defaults for `key_usage` are:
+
+* digitalSignature
+* keyEncipherment
+
+### extended_key_usage
+
+Any valid oid can be used to set one or more `extended_key_usage`.
+In addition to that there is also a list of known aliases that
+will be recognized by the role:
+
+* id-kp-serverAuth
+* id-kp-clientAuth
+* id-kp-codeSigning
+* id-kp-emailProtection
+* id-kp-timeStamping
+* id-kp-OCSPSigning
+* id-kp-ipsecEndSystem
+* id-kp-ipsecTunnel
+* id-kp-ipsecUser
+
+If `extended_key_usage` is not set the role will default to:
+
+* id-kp-serverAuth
+* id-kp-clientAuth
+
+### run hooks
+
+Sometimes you need to execute a command just before a certificate is
+renewed and another command just after. In order to do that use
+`run_before` and `run_after`.
+
+The value provided to `run_before` and `run_after` will be wrapped and
+stored in shell script files that later will be executed by the provider.
+
+
+## CAs and Providers
+
+| CA               | Providers   | CA description                                  | Requirements                                    |
+|------------------|-------------|-------------------------------------------------|-------------------------------------------------|
+| self&#x2011;sign | certmonger* | Issue self-signed certificates from a local CA. |                                                 |
+| ipa              | certmonger* | Issue certificates using the FreeIPA CA.        | Host needs to be enrolled in a FreeIPA server.  |
+
+ *\* Default provider.*
+
+CA represents the CA certificates that will be used to issue and sign the
+requested certificate. Provider it's the method used to send the certificate
```

Comment:
Provider **has** ???

Reply:
@nhosoi "Provider is" was probably intended.

Reply:
I used "represents" to be consistent with the usage in the previous sentence.